src/core/services/compiler/compiler.spec.js

```diff
@@ -163,21 +163,27 @@ describe('$mdCompiler service', function() {
         var called = false;
         var data = compile({
           template: 'hello',
-          controller: function($scope) {
-            expect(this.name).toBe('Bob');
-            expect($scope.$apply).toBeTruthy(); // test DI working properly
-            called = true;
-          },
+          controller: TestController,
           controllerAs: 'ctrl',
           bindToController: true,
           locals: { name: 'Bob' }
         });
+
         var scope = $rootScope.$new();
         data.link(scope);
         expect(scope.ctrl.name).toBe('Bob');
         expect(called).toBe(true);
-      }));
 
+
+        function TestController($scope) {
+          called = true;
+          expect($scope.$apply).toBeTruthy(); // Ensure that DI works properly
+        }
+
+        TestController.prototype.$onInit = function() {
+          expect(this.name).toBe('Bob');
```

Comment:
For example, for the constructor you check that `called` has been set to true, so you know that the constructor has been called (and the expectations it contains verified).
There is nothing similar for `$onInit()`.

You could add `expect('foo').toBe('bar')` inside `$onInit()` and the test would not fail if `$onInit()` is never called. This means that the test may not catch some future regression.